src/java.base/share/classes/javax/crypto/JceSecurity.java.template

```diff
@@ -235,7 +235,16 @@ final class JceSecurity {
 
     // return whether this provider is properly signed and can be used by JCE
     static boolean canUseProvider(Provider p) {
-        return getVerificationResult(p) == null;
+        Exception e = getVerificationResult(p);
+        if (debug != null) {
+            if(e != null) {
```

Comment:
line 239 and 240 can be combined into one line?

Reply:
Also, may just rely on the toString() impl instead of separate handling of getClass() and getMessage()?